.github/workflows/ci-e2e-badger.yaml

```diff
@@ -40,6 +40,22 @@ jobs:
             STORAGE=badger make jaeger-v2-storage-integration-test
             ;;
         esac
+          
+    - name: Upload Metrics Artifact
```

Comment:
Suggestion: these actions are very repetitive in each of the workflows. I would suggest extracting them into a helper action under .githib/actions